Base/QTGUI/qSlicerApplication.cxx

```diff
@@ -396,21 +396,24 @@ bool qSlicerApplication::notify(QObject *receiver, QEvent *event)
   catch( std::exception& exception )
     {
     QString errorMessage;
-    errorMessage = tr("%1 has caught an internal error.\n\n").arg(this->applicationName());
-    errorMessage += tr("You may be able to continue from this point, but results are undefined.\n\n");
-    errorMessage += tr("Suggested action is to save your work and restart.");
-    errorMessage += tr("\n\nIf you have a repeatable sequence of steps that causes this message, ");
+    errorMessage = tr("%1 has caught an exception, ").arg(this->applicationName());
+    errorMessage += tr("please save your work and restart.\n\n");
+    errorMessage += tr("If you have a repeatable sequence of steps that causes this message, ");
     errorMessage += tr("please report the issue following instructions available at http://slicer.org\n\n\n");
     errorMessage += tr("The message detail is:\n\n");
     errorMessage += tr("Exception thrown in event: ") + exception.what();
+    if (!tr("bad allocation").compare(exception.what()) || !tr("std::bad_alloc").compare(exception.what()))
+      {
+        errorMessage += tr("\n\nIncreasing system memory may resolve this issue\n\n");
+      }
     qCritical() << errorMessage;
     if (this->commandOptions()->isTestingEnabled())
       {
       this->exit(EXIT_FAILURE);
       }
     else
       {
-      QMessageBox::critical(this->mainWindow(),tr("Internal Error"), errorMessage);
+      QMessageBox::critical(this->mainWindow(),tr("System Exception"), errorMessage);
```

Comment:
Users don't know what "Exception" means. I think "Internal error" was not too bad - maybe not entirely fair, as Slicer kind of took the blame for something that might have been caused by a bad workflow. Maybe we can change it to "Application error"?

Reply:
Subject and message body were changed to "Application error".